odoo/addons/base/models/ir_config_parameter.py

```diff
@@ -56,6 +74,60 @@ def init(self, force=False):
             if force or not params:
                 params.set_param(key, func())
 
+    @api.model
+    def get(self, key) -> [str, bool, int, float, None]:
+        self.check_access('read')
+        return self._get(key)
+
+    @ormcache('key')
+    def _get(self, key, warning=True):
+        # we bypass the ORM because get_param() is used in some field's depends,
+        # and must therefore work even when the ORM is not ready to work
+        self.flush_model()
+        self.env.cr.execute(SQL('SELECT "value", "type" FROM ir_config_parameter WHERE key = %s', key))
+        result = self.env.cr.fetchone()
+        if not result:
+            if warning and key not in USED_BEFORE_DEFINED:
+                USED_BEFORE_DEFINED.add(key)
+                _logger.warning(f'ir.config_parameter used before defined: {key}')
```

Comment:
Use warnings module instead. It will warn only once (and you won't need USED_BEFORE_DEFINED).